a3fe/run/leg.py

```diff
@@ -803,19 +805,10 @@ def setup_stages(
                     _shutil.copy(
                         restraint_file, f"{stage_input_dir}/restraint_{i + 1}.txt"
                     )
-            # Create a new config for this stage
-            stage_config = self.engine_config.copy()
-            
-            # Set configuration options
-            stage_config.perturbed_residue_number = perturbed_resnum
-            stage_config.use_boresch_restraints = self.leg_type == _LegType.BOUND
-            stage_config.turn_on_receptor_ligand_restraints = self.leg_type == _LegType.BOUND
-            stage_config.charge_difference = -lig_charge  # Use co-alchemical ion approach when there is a charge difference
-            
             # Set lambda values from default dictionary
-            stage_config.lambda_array = stage_config.default_lambda_values[self.leg_type.config_key][stage_type.config_key]
+            self.engine_config.lambda_array = self.engine_config.default_lambda_values[self.leg_type][stage_type]
```

Comment:
You'll need to create the stage_config before this (like you had before) so that you can supply different lambda values for each stage.